app/page.tsx

```diff
@@ -74,6 +74,11 @@ async function getASRouterLocalColumnFromJSON(
     channel,
   );
 
+  // Microsurvey check
+  if (fxmsMsgInfo.id.toLowerCase().includes("survey")) {
+    fxmsMsgInfo.isMicrosurvey = true;
+  }
```

Comment:
This feels like it probably wants to be in a `messageHasMicrosurvey` function in messageUtils.ts to keep the implementation details like message parsing heuristics together with message-related functions rather than in page.tsx.  Over time, I think there's a bunch of things that we probably want to move to their own files, probably including the JSON reading stuff.  No rush, however.